packages/sw-runtime-caching/src/lib/request-wrapper.js

```diff
@@ -184,11 +181,11 @@ class RequestWrapper {
       assert.isInstance({request}, Request);
     }
 
-    const clonedRequest = this.pluginCallbacks.fetchDidFail ?
+    const clonedRequest = this.plugins.has('fetchDidFail') ?
```

Comment:
This is @prateekbh's code from https://github.com/GoogleChrome/sw-helpers/pull/260, but from what I recall from that review, it's necessary to clone the request early, before the request might be modified by a  `requestWillFetch` plugin. We need to save a copy of the original request, not the modified request, to be potentially used by a `fetchDidFail` plugin.

Looking over the code again, I think the cloning is happening in the correct place. I can throw in some additional comments to clarify the logic.